File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem31_ReverseLinkedListTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem29_DeleteTheMiddleNodeOfALinkedList.ListNode;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem31_ReverseLinkedListTest {

    private final Problem31_ReverseLinkedList test = new Problem31_ReverseLinkedList();

    @Test
    void reverseList() {
        ListNode head = new ListNode(1);
        head.next = new ListNode(2);
        head.next.next = new ListNode(3);
        head.next.next.next = new ListNode(4);
        head.next.next.next.next = new ListNode(5);

        assertEquals(5, test.reverseList(head).val);
    }
}

File: JavaScript/leetcode75/src/test/Problem31_ReverseLinkedList.test.js
const reverseList = require("../main/Problem31_ReverseLinkedList");

function ListNode(val, next) {
  this.val = val === undefined ? 0 : val;
  this.next = next === undefined ? null : next;
}

test("should return reverse list", () => {
  let list = new ListNode(1, 2);
  list.next = new ListNode(2, 3);
  list.next.next = new ListNode(3, 4);
  list.next.next.next = new ListNode(4, 5);
  expect(reverseList(list).val).toBe(5);
});
